module/os/linux/zfs/zfs_acl.c

```diff
@@ -2468,6 +2468,8 @@ zfs_zaccess_trivial(znode_t *zp, uint32_t *working_mode, cred_t *cr,
 	defined(HAVE_IOPS_PERMISSION_IDMAP))
 	if (mnt_ns)
 		err = generic_permission(mnt_ns, ZTOI(zp), mask);
+	else
+		err = 0;
```

Comment:
Looking at this in hindsight, it occurred to me that setting 0 when a pointer is NULL in a permission check might be the wrong thing to do. However, on Linux, mnt_ns should never be NULL, so this is really just to suppress the compiler warning.

Reply:
I thought I have removed the "if" line, but apparently it was overlooked. Yes mnt_ns should never be NULL.